mtest/libmscore/text/tst_text.cpp

```diff
@@ -763,6 +765,48 @@ void TestText::testDropUnicodeAfterSMUFLwhenCursorSetToSymbol()
       QCOMPARE(text->xmlText(), QString("<sym>noteheadWhole</sym>𝄎"));
       }
 
+//---------------------------------------------------------
+///   testDropBasicUnicodeWhenNotInEditMode
+///     Simple test dropping basic unicode, but excercising the path when edit mode not already engaged
+//---------------------------------------------------------
+
+void TestText::testDropBasicUnicodeWhenNotInEditMode()
+      {
+      Text* text = new Text(score);
+      text->initSubStyle(SubStyle::DYNAMICS);
+      text->setPlainText(QString(""));
+      text->layout();
+
+      DropData dropFSymbol;
+      FSymbol* fsymbol = new FSymbol(score);
+      fsymbol->setCode(0x4D); // Basic Unicode code for 'M'
+      dropFSymbol.element = fsymbol;
+      text->drop(dropFSymbol);
+
+      QCOMPARE(text->xmlText(), QString("M"));
+      }
+
+//---------------------------------------------------------
+///   testDropSupplementaryUnicodeWhenNotInEditMode
+///     Simple test dropping supplementary unicode, but excercising the path when edit mode not already engaged
+//---------------------------------------------------------
+
+void TestText::testDropSupplementaryUnicodeWhenNotInEditMode()
+      {
+      Text* text = new Text(score);
+      text->initSubStyle(SubStyle::DYNAMICS);
+      text->setPlainText(QString(""));
+      text->layout();
+
+      DropData dropFSymbol;
+      FSymbol* fsymbol = new FSymbol(score);
+      fsymbol->setCode(0x0001D10E); // Supplementary Unicode code for '𝄎'
+      dropFSymbol.element = fsymbol;
+      text->drop(dropFSymbol);
+
+      QCOMPARE(text->xmlText(), QString("𝄎"));
+      }
```

Comment:
So this test was failing, because earlier code would truncate the 32-bit unicode into a 16-bit unicode char, and only insert that, when not in edit mode.